azurerm/internal/services/web/resource_arm_app_service_virtual_network_swift_connection_slot.go

```diff
@@ -0,0 +1,216 @@
+	if err != nil {
+		return fmt.Errorf("Error parsing Azure Resource ID %q", subnetID)
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["sites"]
+	subnetName := subnetID.Path["subnets"]
+	virtualNetworkName := subnetID.Path["virtualNetworks"]
+	slotName := d.Get("slot_name").(string)
+
+	locks.ByName(virtualNetworkName, network.VirtualNetworkResourceName)
+	defer locks.UnlockByName(virtualNetworkName, network.VirtualNetworkResourceName)
+
+	locks.ByName(subnetName, network.SubnetResourceName)
+	defer locks.UnlockByName(subnetName, network.SubnetResourceName)
+
+	appServiceExists, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(appServiceExists.Response) {
+			return fmt.Errorf("Error retrieving existing App Service %q (Resource Group %q): App Service not found in resource group", name, resourceGroup)
+		}
+		return fmt.Errorf("Error retrieving existing App Service %q (Resource Group %q): %s", name, resourceGroup, err)
+	}
+
+	slotExists, err := client.GetSlot(ctx, resourceGroup, name, slotName)
+	if err != nil {
+		if utils.ResponseWasNotFound(slotExists.Response) {
+			return fmt.Errorf("Error retrieving existing App Service Slot %q (App Service %q / Resource Group %q): App Service not found in resource group", slotName, name, resourceGroup)
+		}
+		return fmt.Errorf("Error retrieving existing App Service Slot %q (App Service %q / Resource Group %q): %s", slotName, name, resourceGroup, err)
+	}
+
+	connectionEnvelope := web.SwiftVirtualNetwork{
+		SwiftVirtualNetworkProperties: &web.SwiftVirtualNetworkProperties{
+			SubnetResourceID: utils.String(d.Get("subnet_id").(string)),
+		},
+	}
+	if _, err = client.CreateOrUpdateSwiftVirtualNetworkConnectionSlot(ctx, resourceGroup, name, connectionEnvelope, slotName); err != nil {
+		return fmt.Errorf("Error creating/updating App Service Slot VNet association between %q (App Service %q / Resource Group %q) and Virtual Network %q: %s", slotName, name, resourceGroup, virtualNetworkName, err)
+	}
+
+	read, err := client.GetSwiftVirtualNetworkConnectionSlot(ctx, resourceGroup, name, slotName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving App Service Slot VNet association between %q (App Service %q / Resource Group %q) and Virtual Network %q: %s", slotName, name, resourceGroup, virtualNetworkName, err)
+	}
+	d.SetId(*read.ID)
+
+	return resourceArmAppServiceVirtualNetworkSwiftConnectionSlotRead(d, meta)
+}
+
+func resourceArmAppServiceVirtualNetworkSwiftConnectionSlotRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing Azure Resource ID %q", id)
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["sites"]
+	slotName := id.Path["slots"]
```

Comment:
Could we write a parse ID function like can be seen in #5779

Reply:
done

Reply:
I've also implemented the change for `resource_arm_app_service_virtual_network_switch_connect.go`.